lib/renaissance/helpers/constraint.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Renaissance.Helpers.Constraint do
+  import Ecto.Changeset
+  alias Renaissance.Helpers.Adapt
+  alias Renaissance.Auctions
+
+  def amount_constraint(changeset, field) do
+    auction_id = get_change(changeset, :auction_id, 0)
+
+    proposed = get_change(changeset, field, 0) |> Adapt.money_value()
+    current = Auctions.get_current_amount(auction_id) |> Adapt.money_value()
+
+    if proposed > current do
```

Comment:
This check is not guaranteed to be accurate with concurrent bids being placed. Is that an issue? And if so, how could you ensure that a valid bid must always be placed higher than the current highest bid?

Reply:
I don't think it's an issue, at least based on my current understanding, which is as follows:
- If there are concurrent [valid] bids of difference amounts, then (for all practical purposes) it's as if the users' with the lower bids were out-bid, and therefore no special handling is needed. 
-  I updated `get_highest_bid` ([updated query](https://github.com/xray/Renaissance/blob/845f45ee2e/lib/renaissance/bids/bids.ex#L30);  [test](https://github.com/xray/Renaissance/blob/845f45ee2e/test/renaissance/bids/bids_test.exs#L117)) to return the earliest (or at least first recorded) bid if there are multiple valid bids with the highest amount for a given auction so that, if the auction closes in this state, the winner will be the first person to place the bid.